services/wallet/router/router.go

```diff
@@ -147,41 +163,46 @@ func newSuggestedRoutes(
 	return suggestedRoutes, allRoutes
 }
 
+func sendRouterResult(uuid string, result interface{}, err error) {
+	routesResponse := responses.RouterSuggestedRoutes{
+		Uuid: uuid,
+	}
+
+	if err != nil {
+		errorResponse := errors.CreateErrorResponseFromError(err)
+		routesResponse.ErrorResponse = errorResponse.(*errors.ErrorResponse)
+	}
+
+	if suggestedRoutes, ok := result.(*SuggestedRoutes); ok && suggestedRoutes != nil {
+		routesResponse.Best = suggestedRoutes.Best
+		routesResponse.Candidates = suggestedRoutes.Candidates
+		routesResponse.TokenPrice = &suggestedRoutes.TokenPrice
+		routesResponse.NativeChainTokenPrice = &suggestedRoutes.NativeChainTokenPrice
+	}
+
+	signal.SendWalletEvent(signal.SuggestedRoutes, routesResponse)
+}
+
 func (r *Router) SuggestedRoutesAsync(input *requests.RouteInputParams) {
 	r.scheduler.Enqueue(routerTask, func(ctx context.Context) (interface{}, error) {
 		return r.SuggestedRoutes(ctx, input)
 	}, func(result interface{}, taskType async.TaskType, err error) {
-		routesResponse := responses.RouterSuggestedRoutes{
-			Uuid: input.Uuid,
-		}
-
-		if err != nil {
-			errorResponse := errors.CreateErrorResponseFromError(err)
-			routesResponse.ErrorResponse = errorResponse.(*errors.ErrorResponse)
-		}
-
-		if suggestedRoutes, ok := result.(*SuggestedRoutes); ok && suggestedRoutes != nil {
-			routesResponse.Best = suggestedRoutes.Best
-			routesResponse.Candidates = suggestedRoutes.Candidates
-			routesResponse.TokenPrice = &suggestedRoutes.TokenPrice
-			routesResponse.NativeChainTokenPrice = &suggestedRoutes.NativeChainTokenPrice
-		}
-
-		signal.SendWalletEvent(signal.SuggestedRoutes, routesResponse)
+		sendRouterResult(input.Uuid, result, err)
 	})
 }
 
 func (r *Router) StopSuggestedRoutesAsyncCalcualtion() {
+	r.unsubscribeFeesUpdateAccrossAllChains()
 	r.scheduler.Stop()
 }
 
-func (r *Router) SuggestedRoutes(ctx context.Context, input *requests.RouteInputParams) (*SuggestedRoutes, error) {
-	testnetMode, err := r.rpcClient.NetworkManager.GetTestNetworksEnabled()
-	if err != nil {
-		return nil, errors.CreateErrorResponseFromError(err)
-	}
+func (r *Router) StopSuggestedRoutesCalcualtion() {
```

Comment:
typo `Calcualtion` here and the one above. I guess we can keep it at the API level for client compatibility, but no need to keep it here :D

Reply:
@dlipicar updated in both places, better now than later. ;)